python/lsst/ts/scheduler/scheduler_csc.py

```diff
@@ -318,23 +361,23 @@ async def begin_enable(self, data):
 
         if self.simulation_mode == SchedulerModes.SIMULATION:
             self.log.debug("Running in simulation mode. No target production loop.")
-            self.target_production_task = None
+            self._tasks["target_production_task"] = None
 
         elif self.parameters.mode == "SIMPLE":
 
-            self.target_production_task = asyncio.create_task(
+            self._tasks["target_production_task"] = asyncio.create_task(
                 self.simple_target_production_loop()
             )
 
         elif self.parameters.mode == "ADVANCE":
 
-            self.target_production_task = asyncio.create_task(
+            self._tasks["target_production_task"] = asyncio.create_task(
                 self.advance_target_production_loop()
             )
 
         elif self.parameters.mode == "DRY":
```

Comment:
Maybe this `elif` can be combined with the `if` because they both set `target_production_task` to None? The difference is only the `log.debug` statement which can be made generic using

`self.log.debug(f"Running in mode {self.simulation_mode.name}. No target production loop.")`

This way some duplicate code gets removed.